packages/components/pie-checkbox-group/src/index.ts

```diff
@@ -1,20 +1,113 @@
-import { LitElement, html, unsafeCSS } from 'lit';
-import { RtlMixin, defineCustomElement } from '@justeattakeaway/pie-webc-core';
-
+import {
+    LitElement, html, unsafeCSS, PropertyValues,
+} from 'lit';
+import { property } from 'lit/decorators.js';
+import {
+    RtlMixin,
+    defineCustomElement,
+    FormControlMixin,
+    validPropertyValues,
+} from '@justeattakeaway/pie-webc-core';
+import { ifDefined } from 'lit/directives/if-defined.js';
 import styles from './checkbox-group.scss?inline';
-import { CheckboxGroupProps } from './defs';
+import {
+    ON_CHECKBOX_GROUP_DISABLED,
+    CheckboxGroupProps,
+    defaultProps,
+    statusTypes,
+} from './defs';
+import '@justeattakeaway/pie-assistive-text';
 
 // Valid values available to consumers
 export * from './defs';
 
 const componentSelector = 'pie-checkbox-group';
+const assistiveTextId = 'assistive-text';
 
 /**
  * @tagname pie-checkbox-group
+ * @event {CustomEvent} pie-checkbox-group-disabled - triggered after the disabled state of the checkbox group changes.
  */
-export class PieCheckboxGroup extends RtlMixin(LitElement) implements CheckboxGroupProps {
+export class PieCheckboxGroup extends FormControlMixin(RtlMixin(LitElement)) implements CheckboxGroupProps {
+    @property({ type: String })
+    public name?: CheckboxGroupProps['name'];
+
+    @property({ type: String })
+    public label?: CheckboxGroupProps['label'];
+
+    @property({ type: String })
+    public assistiveText?: CheckboxGroupProps['assistiveText'];
+
+    @property({ type: String })
+    @validPropertyValues(componentSelector, statusTypes, defaultProps.status)
+    public status = defaultProps.status;
+
+    @property({ type: Boolean, reflect: true })
+    public disabled = defaultProps.disabled;
+
+    get _slot () {
+        return this.shadowRoot?.querySelector('slot');
+    }
+
+    get _slottedChildren () {
```

Comment:
I think we could use the [`queryAssignedElements` decorator](https://lit.dev/docs/api/decorators/#queryAssignedElements) for this? Happy for you to do this in the next PR if you like

Reply:
And maybe `@query` would work for `_slot` too

Reply:
As a bonus, I think `queryAssignedElements` returns an array so you might not need to spread it below

Reply:
👌 this is perfect